android/src/main/java/it/innove/Peripheral.java

```diff
@@ -391,73 +407,74 @@ public void onReadRemoteRssi(BluetoothGatt gatt, int rssi, int status) {
     }
 
     private void setNotify(UUID serviceUUID, UUID characteristicUUID, Boolean notify, Callback callback) {
-        if (!isConnected()) {
-            callback.invoke("Device is not connected", null);
-            return;
-        }
         Log.d(LOG_TAG, "setNotify");
+        Boolean setNotifyCompleted = false;
```

Comment:
💁 I'm using setNotifyCompleted so check if we successfully called `gatt.writeDescriptor`. If not, we will want to complete the command at the end of this method.